File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.java
import java.util.stream.*;
import java.util.*;

class Solution {
    public int lcm(int a, int b){
        int sourceMin = Math.min(a, b);
        int sourceMax = Math.max(a, b);
        int min = sourceMin;
        int max = sourceMax;
        
        while(max != min){
            if (min < max) min += sourceMin;
            else max += sourceMax;
        }
        
        return max;
    }
    
    public int solution(int[] arr) {
        int temp = 1;
        for(int i : arr){
            temp = lcm(temp, i);
        }
        return temp;
    }
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.js
function lcm(a1, b1){
    let a = Math.max(a1,b1);
    let max = a;
    let b = a1 + b1 - a;
    while (a % b !== 0){
        a += max;
    }
    
    return a;
}

function solution(arr) {
    return arr.reduce((acc, v) => lcm(acc,v), 1);
}